lib/bandit/http1/handler.ex

```diff
@@ -88,6 +100,35 @@ defmodule Bandit.HTTP1.Handler do
     end
   end
 
+  defp maybe_upgrade_h2c(state, req, transport_info, method, request_target, headers) do
+    with {:http_2_enabled, true} <- {:http_2_enabled, state.http_2_enabled},
+         {:upgrade, "h2c"} <- {:upgrade, Bandit.Headers.get_header(headers, "upgrade")},
+         %Bandit.TransportInfo{secure?: false} <- transport_info,
+         {:ok, connection_headers} <- Bandit.Headers.get_connection_header_keys(headers),
+         {:settings, [{"http2-settings", settings_payload}]} <-
+           {:settings, Enum.filter(headers, fn {key, _value} -> key == "http2-settings" end)},
+         {:ok, settings_payload} <- Base.url_decode64(settings_payload),
+         {:ok, body, req} <- Bandit.HTTP1.Adapter.read_req_body(req, []),
```

Comment:
To handle the initial request, we need to read the request body in HTTP/1.1 (as per RFC7540§3.2)

To do that I am using `Bandit.HTTP1.Adapter.read_req_body/2` and used an empty list for the options. I don't like this but didn't find a better solution. One problem with this is the read_body options will be ignored (e.g. read_timeout and read_length).

Reply:
This is actually more than a bit tricky as we need a bit of logic to handle `{:more,...}` and other tuples. Something like:

```
{:ok, body, req} <- do_read_req_body(req)

.....

# This function is only used during the h2c upgrades
defp do_read_req_body(req, acc \\ <<>>) do
  case Bandit.HTTP1.Adapter.read_req_body(req, []) do
    {:ok, body, req} -> {:ok, acc <> body, req}
    {:more, body, req} -> do_read_req_body(req, acc <> body)
    {:error, error} -> {:error, error}
  end
end 
```